### PR TITLE
[chore] Prevent actions from cancelling on main

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
In an effort to ensure all commits into main run to completion this PR updates several actions to ignore skipping on the main branch.  The technique is the same used to ensure build-and-test and build-and-test-windows never get skipped on main.

### CODE REVIEW

File: .github/workflows/auto-assign-owners.yml [hunk 1]
```diff
@@ -3,8 +3,9 @@ on:
   pull_request_target:
     types: [opened, ready_for_review]
 
+# Do not cancel this workflow on main.
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref }}
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```
Comment (codeboten) [inline .github/workflows/auto-assign-owners.yml:8]: I don't think we need this change as this job doesn't run on main

File: .github/workflows/changelog.yml [hunk 2]
```diff
@@ -16,8 +16,9 @@ env:
   # Cache downloads for this workflow consistently run in under 1 minute
   SEGMENT_DOWNLOAD_TIMEOUT_MINS: 5
 
+# Do not cancel this workflow on main. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16616
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref }}
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```
Comment (codeboten) [inline .github/workflows/changelog.yml:21]: I don't think we need this change as this job doesn't run on main

File: .github/workflows/auto-assign-owners.yml (hunk 1 above)
Comment (TylerHelmuth) [reply .github/workflows/auto-assign-owners.yml]: Good catch

File: .github/workflows/changelog.yml (hunk 2 above)
Comment (TylerHelmuth) [reply .github/workflows/changelog.yml]: Good catch

**Review by codeboten — APPROVED** [review]

**Pull request merged**